src/services/multiversx-communication/mx.proxy.service.ts

```diff
@@ -281,4 +347,24 @@ export class MXProxyService {
         MXProxyService.smartContracts[key] = newSC;
         return newSC;
     }
+
+    private async createSmartContractTransactionsFactory(
+        contractAbiPath: string,
+        contractInterface: string,
+        chainID: string,
+    ): Promise<SmartContractTransactionsFactory> {
+        const jsonContent: string = await promises.readFile(contractAbiPath, {
```

Comment:
maybe try to read from `smartContractTransactionFactories` before recomputing it?

Reply:
`createSmartContractTransactionsFactory` is a private method that only creates a `SmartContractTransactionsFactory` from an abi. A check if performed in the `getSmartContractTransactionFactory` method before calling this.